src/main/java/org/breedinginsight/brapps/importer/daos/BrAPIObservationUnitDAO.java

```diff
@@ -113,9 +120,47 @@ public List<BrAPIObservationUnit> getObservationUnitsForTrialDbId(@NotNull UUID
                 .getProgramDbId()));
         observationUnitSearchRequest.trialDbIds(List.of(trialDbId));
 
-        ObservationUnitsApi api = brAPIEndpointProvider.get(programDAO.getCoreClient(program.getId()), ObservationUnitsApi.class);
-        return brAPIDAOUtil.search(api::searchObservationunitsPost,
+        return searchObservationUnitsAndProcess(observationUnitSearchRequest, programId);
+    }
+
+
+    /**
+     * Perform observation unit search and process returned observation units to handle any modifications to the data
+     * to be returned by bi-api
+     */
+    private List<BrAPIObservationUnit> searchObservationUnitsAndProcess(BrAPIObservationUnitSearchRequest request, UUID programId) throws ApiException {
+
+        ObservationUnitsApi api = brAPIEndpointProvider.get(programDAO.getCoreClient(programId), ObservationUnitsApi.class);
+        List<BrAPIObservationUnit> brapiObservationUnits = brAPIDAOUtil.search(api::searchObservationunitsPost,
                 api::searchObservationunitsSearchResultsDbIdGet,
-                observationUnitSearchRequest);
+                request);
+
+        processObservationUnits(brapiObservationUnits);
+        return brapiObservationUnits;
+    }
+
+    private void processObservationUnits(List<BrAPIObservationUnit> brapiObservationUnits) {
+
+        // if has treatments in additionalInfo but not treatments property, copy to treatments property
```

Comment:
The code doesn't match this comment fully.  Looks like the code checks if there's the treatments property in the additionalInfo, and if so, replace whatever could be in the `Treatments` field of the OU.  Doesn't look like it's checking to see if there is any value in the `Treatments` field in the OU before calling `setTreatments`. Is this the desired logic?

Reply:
Updated comment to match code, intent is to just overwrite the treatments property if one exists in the additionalInfo